### ISSUE TITLE
SC2015: Warning disappears when there is a continuation

### ISSUE THREAD

**Issue by nishantvarma** [koalaman/shellcheck#2314]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2015 (A && B || C)
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

caching=true

if $caching; then
        cache-create && cache-update || true
fi && echo "done" 

```

#### Here's what shellcheck currently says:

`No issues detected!`

#### Here's what I wanted or expected to see:

`SC2015: Note that A && B || C is not if-then-else. C may run when A is true.`

Note that it does show the warning when you remove the `&& echo "done"` block.

```sh

#!/bin/bash

caching=true

if $caching; then
        cache-create && cache-update || true
fi

```

**Comment by koalaman** [comment]
Neat case. ShellCheck does not warn when `a && b || c` is a condition, such as `if a && b || c; then ...` since then the point is presumably not to act as an if-then-else by itself but rather to evaluate some condition. That is also what happens here, in a more roundabout way, so in that sense it's working as intended.

Arguably ShellCheck should warn when the construct is in a condition but whose exit code is discarded, like `if true; then a && b || c; d; fi && echo "done"` but the infrastructure is currently lacking for such a niche case.
